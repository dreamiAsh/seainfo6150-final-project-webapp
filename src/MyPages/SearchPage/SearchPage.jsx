import React, {useEffect, useState} from "react";
import Styles from "./SearchPage.module.css"
import RecipeList from "../../Components/RecipeList/RecipeList"

const SearchPage = () => {

    const [clicked, setClicked] = useState();
    function onClick(e) {
        setClicked(true)
    }

    return (
        clicked ? 
        (<div>
            <div className={Styles.box}>
                <input className={Styles.searchInput} type="text" placeholder="  Enter a recipe name"/>
                <p className={Styles.searchButton} onClick={onClick}>Search</p>
            </div>
            <RecipeList/>
        </div>) 
        :
        (<div className={Styles.box}>
            <input className={Styles.searchInput} type="text" placeholder="  Enter a recipe name"/>
            <p className={Styles.searchButton} onClick={onClick}>Search</p>
        </div>)
    )
}

export default SearchPage
