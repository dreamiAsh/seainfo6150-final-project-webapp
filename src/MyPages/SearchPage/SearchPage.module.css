.box{
    padding-top: 160px;
    width: 100%;
}

.searchInput{
    width: 55%;
    height: 32px;
    float: left;
    margin-bottom: 200px;
    margin-left: 17%;
}
.searchButton{
    width: 10%;
    height: 35px;
    background: orange;
    text-align: center;
    line-height: 35px;
    float: left;
    margin-bottom: 200px;
}