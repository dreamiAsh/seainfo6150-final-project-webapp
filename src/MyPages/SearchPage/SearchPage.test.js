import React from "react";
import { render, fireEvent } from "@testing-library/react";
import SearchPage from "./SearchPage.jsx";

it("renders the SearchPage component correctly", () => {
  const { container } = render(<SearchPage />);
  expect(container).toMatchSnapshot();
});