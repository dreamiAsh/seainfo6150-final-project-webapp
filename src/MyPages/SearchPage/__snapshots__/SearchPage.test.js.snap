// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the SearchPage component correctly 1`] = `
<div>
  <div
    class="box"
  >
    <input
      class="searchInput"
      placeholder="  Enter a recipe name"
      type="text"
    />
    <p
      class="searchButton"
    >
      Search
    </p>
  </div>
</div>
`;
